src/NServiceBus.AcceptanceTests/Routing/SubscriptionBehavior.cs

```diff
@@ -36,16 +44,18 @@ public SubscriptionBehavior(Action<SubscriptionEventArgs, TContext> action, TCon
 
         static string GetSubscriptionMessageTypeFrom(IncomingMessage msg)
         {
-            return (from header in msg.Headers where header.Key == Headers.SubscriptionMessageType select header.Value).FirstOrDefault();
+            string headerValue;
+            return msg.Headers.TryGetValue(Headers.SubscriptionMessageType, out headerValue) ? headerValue : null;
         }
 
         Action<SubscriptionEventArgs, TContext> action;
         TContext scenarioContext;
+        MessageIntentEnum intentToHandle;
 
         internal class Registration : RegisterStep
         {
-            public Registration()
-                : base("SubscriptionBehavior", typeof(SubscriptionBehavior<TContext>), "So we can get subscription events")
+            public Registration(string id, Func<IBuilder, IBehavior> behaviorFactory)
+                : base(id, typeof(SubscriptionBehavior<TContext>), "notify subscription events", behaviorFactory)
             {
                 InsertBeforeIfExists("ProcessSubscriptionRequests");
```

Comment:
while it would be possible, currently the overloads taking a behavior factory do not accept an id. Since we're using the same behavior with different parameters for subscribe/unsubscribe, it requires different ids in case you want to use both behaviors when using both in the same test. So I'd say let's keep the registrations for now.